testsuite/tests/compiler-libs/test_untypeast.ml

```diff
@@ -37,3 +37,12 @@ run {| fun x y z -> (function w -> x y z w) |};;
 [%%expect{|
 - : string = "fun x y z -> (function | w -> x y z w)"
 |}];;
+
+(***********************************)
+(* Untypeast/pprintast correctly handle value binding type annotations. *)
+
+run {| let foo : 'a. 'a -> 'a = fun x -> x in foo |};;
```

Comment:
The parser also supports `let foo : type a . a -> a = fun x -> x`, does this work as expected?

Reply:
This syntax is elaborated in Typecore to
```ocaml
let (foo: 'a. 'a -> 'a) = fun (type a) -> ( (fun x -> x) : 'a -> 'a)
```
and it is thus is similar to the first form in the typedtree.